### PR TITLE
[Merged by Bors] - chore: more `refine'` to `refine`

### PR DESCRIPTION
**Pull request by adomani**
A PR analogous to #13166.  The reason some of these `refine'` are now caught is that my previous script missed all instances.  I am not sure why it is still missing other `refine'`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by adomani** [comment]
!bench

File: Mathlib/Algebra/Order/Module/OrderedSMul.lean [hunk 1]
```diff
@@ -121,7 +121,7 @@ theorem OrderedSMul.mk' (h : ∀ ⦃a b : M⦄ ⦃c : 𝕜⦄, a < b → 0 < c 
     refine fun a b c hab hc => (h hab hc).lt_of_ne ?_
     rw [Ne, hc.ne'.isUnit.smul_left_cancel]
     exact hab.ne
-  refine' { smul_lt_smul_of_pos := fun {a b c} => hlt' a b c..}
+  refine' { smul_lt_smul_of_pos := fun {a b c} => hlt' a b c.. }
```
Comment (mattrobball) [inline Mathlib/Algebra/Order/Module/OrderedSMul.lean:124]: Is this one removable?

File: Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean [hunk 2]
```diff
@@ -200,7 +200,7 @@ if two direct sums over `s` are isomorphic, then they have the same multipliciti
 theorem equiv_of_iso (o : HomOrthogonal s) {α β : Type} [Finite α] [Finite β] {f : α → ι}
     {g : β → ι} (i : (⨁ fun a => s (f a)) ≅ ⨁ fun b => s (g b)) :
     ∃ e : α ≃ β, ∀ a, g (e a) = f a := by
-  refine' ⟨Equiv.ofPreimageEquiv _, fun a => Equiv.ofPreimageEquiv_map _ _⟩
+  refine' ⟨Equiv.ofPreimageEquiv ?_, fun a => Equiv.ofPreimageEquiv_map _ _⟩
```
Comment (mattrobball) [inline Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean:203]: This one also?

File: Mathlib/NumberTheory/ClassNumber/Finite.lean [hunk 3]
```diff
@@ -298,7 +298,7 @@ theorem exists_mk0_eq_mk0 [IsDedekindDomain S] [Algebra.IsAlgebraic R L] (I : (I
   obtain ⟨b, b_mem, b_ne_zero, b_min⟩ := exists_min abv I
   suffices Ideal.span {b} ∣ Ideal.span {algebraMap _ _ M} * I.1 by
     obtain ⟨J, hJ⟩ := this
-    refine ⟨⟨J, ?_⟩, ?_, ?_⟩
+    refine' ⟨⟨J, _⟩, _, _⟩
```
Comment (mattrobball) [inline Mathlib/NumberTheory/ClassNumber/Finite.lean:301]: Backwards?

File: Mathlib/Algebra/Homology/HomotopyCategory/ShiftSequence.lean [hunk 4]
```diff
@@ -119,8 +119,8 @@ instance {K L : CochainComplex C ℤ} (φ : K ⟶ L) (n : ℤ) [QuasiIso φ] :
     QuasiIso (φ⟦n⟧') where
   quasiIsoAt a := by
     rw [quasiIsoAt_iff_isIso_homologyMap]
-    refine' (NatIso.isIso_map_iff
-      ((homologyFunctor C (ComplexShape.up ℤ) 0).shiftIso n a (n + a) rfl) φ).2 _
+    refine (NatIso.isIso_map_iff
```
Comment (rosborn) [inline Mathlib/Algebra/Homology/HomotopyCategory/ShiftSequence.lean:122]: I feel like this should be `apply` (or perhaps we should discuss deprecating `apply` in favor of `refine`).

File: Mathlib/Algebra/Polynomial/Basic.lean [hunk 5]
```diff
@@ -363,7 +363,7 @@ instance unique [Subsingleton R] : Unique R[X] :=
   { Polynomial.inhabited with
     uniq := by
       rintro ⟨x⟩
-      refine' congr_arg ofFinsupp _
+      refine congr_arg ofFinsupp ?_
```
Comment (rosborn) [inline Mathlib/Algebra/Polynomial/Basic.lean:366]: Also `apply`, although I'm going to stop mentioning it now.

**Comment by adomani** [comment]
Matt, thanks for the review!  Somewhere, I must have done something wrong with a git commit.  I'm about to get into a plane, but will fix/answer later tonight!

**Review by rosborn — APPROVED** [review]
Other than Matt's comments this looks correct.

File: Mathlib/Algebra/Order/Module/OrderedSMul.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Algebra/Order/Module/OrderedSMul.lean:124]: I checked: it isn't.

File: Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean (hunk 2 above)
Comment (grunweg) [reply Mathlib/CategoryTheory/Preadditive/HomOrthogonal.lean:203]: This one, however, is:
```suggestion
  refine ⟨Equiv.ofPreimageEquiv ?_, fun a => Equiv.ofPreimageEquiv_map _ _⟩

File: Mathlib/NumberTheory/ClassNumber/Finite.lean (hunk 3 above)
Comment (grunweg) [reply Mathlib/NumberTheory/ClassNumber/Finite.lean:301]:
```suggestion
    refine ⟨⟨J, ?_⟩, ?_, ?_⟩

Comment (grunweg) [reply Mathlib/NumberTheory/ClassNumber/Finite.lean:301]: The refine' below is *not* removable; could this be the source of the error (that *all* `refine'`s in that proof were reverted)?

Comment (mattrobball) [reply Mathlib/NumberTheory/ClassNumber/Finite.lean:301]: This is not a good sign. Metavariable shenanigans might be happening here

**Comment by grunweg** [comment]
I took the liberty to push a commit replacing a handful of `refine` (in files modified) with apply. Feel free to revert, of you disagree with it.

**Review by grunweg — COMMENTED** [review]
I've read through it all and confirmed that all this PR does is replacing refine' by refine.
(and sometimes `refine stuff ?_` by `apply stuff`, where `stuff` could be several arguments).

(And `Mathlib/Algebra/BigOperators/Group/Multiset.lean`, and comment-only change.)
Benchmarking was neutral (though doesn't include the last batch); if any performance surprises come up, the last commit is easy to revert.

maintainer merge

File: Mathlib/NumberTheory/ClassNumber/Finite.lean (hunk 3 above)
Comment (grunweg) [reply Mathlib/NumberTheory/ClassNumber/Finite.lean]: To follow up on this: this change is now removed from the diff: the diff in that file looks entirely reasonable.

It's very possible meta-variable shenanigans happen here - but my impression is that the current change should not perturb them.

**Comment by jcommelin** [comment]
bors d+

**Comment by riccardobrasca** [comment]
Since bors if free after the current batch.

bors r+ p=37

**Comment by riccardobrasca** [comment]
Mmm, the conflict is solved, isn't it?

**Comment by adomani** [comment]
Should be, let me try again to merge master

**Comment by adomani** [comment]
Yes, I just pulled and merged origin/master with no conflicts.  Should I push again?

**Comment by riccardobrasca** [comment]
I don't see any conflict, so I removed the label by hand.

**Comment by adomani** [comment]
Oh, this is going to be very quick, though, since the cache that it build was already essentially equal to the one on master.  In fact, it is already in the testing phase.

**Comment by adomani** [comment]
Once this is merged, I'll merge master on the other one and push again.
